web/static/js/thesis-editor.js

```diff
@@ -16,35 +16,53 @@ import RawHtmlEditor from './content_types/raw_html_editor'
 import ImageEditor from './content_types/image_editor'
 import TextEditor from './content_types/text_editor'
 
+const thesisContainer = document.querySelector('#thesis-container')
+
 class ThesisEditor extends React.Component {
 
   constructor (props) {
     super(props)
     this.state = {
-      editing: false,
-      pageModified: false,
-      pageToolsHidden: true,
-      trayOpen: false,
-      trayType: null
+      editing:          false,
+      path:             this.pathname(),
+      title:            this.pageTitle(),
+      description:      this.pageDescription(),
+      template:         this.pageTemplate(),
+      templates:        this.pageTemplates(),
+      redirectURL:      this.pageRedirectURL(),
+      pageModified:     false,
+      pageToolsHidden:  true,
+      trayOpen:         false,
+      trayType:         null,
+      deleted:          false
     }
     this.htmlEditor = new HtmlEditor(this)
     this.rawHtmlEditor = new RawHtmlEditor(this)
     this.imageEditor = new ImageEditor(this, {ospryPublicKey: this.ospryPublicKey()})
     this.textEditor = new TextEditor(this)
 
+    this.warnURLRedirect(this.state.redirectURL)
```

Comment:
This is where we pop up an alert saying "This page is set to redirect to /new-page. Follow redirect?" Okay/Cancel.

If they click Okay, we redirect, otherwise we just wait for them to edit the page.